src/upkeep.rs

```diff
@@ -125,32 +125,34 @@ pub async fn do_upkeep(
     }
 
     // 4. Handle processing deadlines
-    if let Ok(processing_count) = store
+    if let Ok(counts) = store
         .handle_processing_deadline()
         .instrument(info_span!("handle_processing_deadline"))
         .await
     {
-        result_context.processing_deadline_reset = processing_count;
+        result_context.processing_deadline_reset = counts.0;
+        result_context.processing_attempts_exceeded = counts.1;
     }
 
-    if let Ok(expired_count) = store
-        .handle_expires_at()
-        .instrument(info_span!("handle_expires_at"))
+    // 5.Handle tasks whos processing_attempts have exceeded max_processing_attempts
+    if let Ok(processing_attempts_exceeded) = store
+        .handle_processing_attempts()
```

Comment:
Now that you're advancing activations to failure inside `handle_processing_deadline` wouldn't this be a no-op?

Reply:
That's true, removed `handle_processing_attempts` in favour for `handle_processing_deadline`.